CONTRIBUTING.md

```diff
@@ -11,7 +11,110 @@ Thank you for your interest in contributing to CUDA Python! Based on the type of
 2. You want to implement a feature, improvement, or bug fix:
     - Please refer to each component's guideline:
        - [`cuda.core`](https://nvidia.github.io/cuda-python/cuda-core/latest/contribute.html)
-       - [`cuda.bindings`](https://nvidia.github.io/cuda-python/cuda-bindings/latest/contribute.html)
+       - [`cuda.bindings`](https://nvidia.github.io/cuda-python/cuda-bindings/latest/contribute.html)¹
+       - [`cuda.pathfinder`](https://nvidia.github.io/cuda-python/cuda-pathfinder/latest/contribute.html)
+
+
+## CI infrastructure overview
+
+The CUDA Python project uses a comprehensive CI pipeline that builds, tests, and releases multiple components across different platforms. This section provides a visual overview of our CI infrastructure to help contributors understand the build and release process.
+
+### CI Pipeline Flow
+
+```
+┌─────────────────────────────────────────────────────────────────────────────────────┐
+│                                   TRIGGER EVENTS                                   │
+│  • Push to main branch              • Pull request                                 │
+│  • Push to pull-request/* branches  • Manual workflow dispatch                    │
+└─────────────────────────────────────────────────────────────────────────────────────┘
+                                             │
+                                             ▼
+┌─────────────────────────────────────────────────────────────────────────────────────┐
+│                                    BUILD STAGE                                     │
+│                                                                                     │
+│  ┌─────────────────┐  ┌─────────────────┐  ┌─────────────────┐                     │
+│  │   linux-64      │  │ linux-aarch64   │  │     win-64      │                     │
+│  │  Self-hosted    │  │  Self-hosted    │  │ GitHub-hosted   │                     │
+│  │ linux-amd64-cpu8│  │ linux-arm64-cpu8│  │  windows-2022   │                     │
```

Comment:
We don't need specific runner names (as they could be changed any time), but we do want to know if it's self- or GH- hosted. Remove the names.